### PR TITLE
add additional tests for coverage,

### PR DESCRIPTION
**Pull request by phlptp**
Add some additional tests to try to get test coverage back to 100%

 refactor the positional parsing to  reduce code duplication

### CODE REVIEW

**Review by henryiii — APPROVED** [review]

**Comment by phlptp** [comment]
@henryiii I am going to merge this in one day unless you object.  (2023-10-7)
There are no API changes only some code refactoring, new tests, and a fix for an edge case on specifically formatted inline comments on flags.  then I will try to tackle the coverage in some of the other files, but this fixes app_inl and config_inl to 100% coverage.

**Comment by henryiii** [comment]
I've already approved so you can merge any time (you don't have to have my approval unless it's potentially controversial or something). And these sorts of cleanups & added tests don't need much review, IMO.

**Pull request merged**
